### PR TITLE
[OCPBUGS-37806] Change RHEL links to new locations

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.16+

Issue:
https://issues.redhat.com/browse/OCPBUGS-37806

Link to docs preview:
- https://81956--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z#installation-user-infra-machines-iso-ibm-z_installing-ibm-z (step 4)
- https://81956--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-lpar#installation-user-infra-machines-iso-ibm-z_installing-ibm-z-lpar (step 4) 
- https://81956--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm#ibm-z-network-connectivity_installing-ibm-z-kvm
- https://81956--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm#installation-bare-metal-config-yaml_installing-ibm-z-kvm
- https://81956--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm#configuring-nbde-static-ip-ibmz-linuxone-environment_installing-ibm-z-kvm

QE review:
- [x] QE has approved this change.

Additional information:
gdoc with old -> new links https://docs.google.com/document/d/17mAfVBhZiFQE6ifQK8tYIbV9EPBOYsGxR7DUUhMKQEw/edit

### CODE REVIEW

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 1]
```diff
@@ -149,7 +149,7 @@ zfcp.allow_lun_scan=0
 Write all options in the parameter file as a single line and make sure you have no newline characters.
 
 ifdef::ibm-z[]
-. Transfer the initramfs, kernel, parameter files, and {op-system} images to z/VM, for example with FTP. For details about how to transfer the files with FTP and boot from the virtual reader, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/sect-installing-zvm-s390[Installing under Z/VM].
+. Transfer the initramfs, kernel, parameter files, and {op-system} images to z/VM, for example with FTP. For details about how to transfer the files with FTP and boot from the virtual reader, see link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/interactively_installing_rhel_over_the_network/index#using-the-zvm-reader_installing-under-z-vm[Booting the installation on {ibm-z-name} to install {op-system-base} in z/VM].
```
Comment (dfitzmau) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:152]: Usually the link metadata would match the link target heading, which is Booting the RHEL installation by using the z/VM Reader.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 2]
```diff
@@ -170,7 +170,7 @@ See link:https://www.ibm.com/docs/en/zvm/latest?topic=commands-ipl[IPL] in IBM D
 +
 endif::ibm-z[]
 ifdef::ibm-z-lpar[]
-. Transfer the initramfs, kernel, parameter files, and {op-system} images to the LPAR, for example with FTP. For details about how to transfer the files with FTP and boot, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_a_standard_rhel_8_installation/installing-in-an-lpar_installing-rhel[Installing in an LPAR].
+. Transfer the initramfs, kernel, parameter files, and {op-system} images to the LPAR, for example with FTP. For details about how to transfer the files with FTP and boot, see link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/interactively_installing_rhel_over_the_network/index#installing-in-an-lpar-using-an-ftp-server_installing-in-an-lpar[Booting the installation on {ibm-z-name} to install {op-system-base} in an LPAR].
```
Comment (dfitzmau) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:173]: Same comment as previous comment.

File: modules/installation-requirements-user-infra-ibm-z-kvm.adoc [hunk 3]
```diff
@@ -57,11 +57,11 @@ To install on {ibm-z-name} under {op-system-base} KVM, you need:
 *   A {op-system-base} KVM host configured with an OSA or RoCE network adapter.
 *   Either a {op-system-base} KVM host that is configured to use bridged networking in libvirt or MacVTap to connect the network to the guests.
 +
-See link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_and_managing_virtualization/index#types-of-virtual-machine-network-connections_configuring-virtual-machine-network-connections[Types of virtual network connections].
+See link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/configuring_and_managing_virtualization/index#virtual-networking-network-address-translation_types-of-virtual-machine-network-connections[Types of virtual network connections].
```
Comment (dfitzmau) [inline modules/installation-requirements-user-infra-ibm-z-kvm.adoc:60]:
```suggestion
See link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/configuring_and_managing_virtualization/index#virtual-networking-network-address-translation_types-of-virtual-machine-network-connections[Types of virtual network machine connections].
```

Comment (dfitzmau) [reply modules/installation-requirements-user-infra-ibm-z-kvm.adoc:60]: The link seems to float on its own but is it relevant to both bullet points or just one? If one, maybe it should be moved to that line.

**Review by dfitzmau — APPROVED** [review]

**Comment by dfitzmau** [comment]
Thanks for tagging me, @max-cx . LGTM as well as all the links render correctly. I added some small suggestions. Feel free to add a peer-review-done label.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc (hunk 1 above)
Comment (max-cx) [reply modules/installation-ibm-z-user-infra-machines-iso.adoc:152]: ACK.

Comment (SNiemann15) [reply modules/installation-ibm-z-user-infra-machines-iso.adoc:152]: @dfitzmau Thanks for spotting this I just realized the toc has a real strange behaviour. When I click on 12.4 once I get the 12.4.1 link, with a second click on 12.4 I get the correct one 
image

**Review by bergerhoffer — COMMENTED** [review]
LGTM for merging!

File: modules/installation-ibm-z-user-infra-machines-iso.adoc (hunk 1 above)
Comment (dfitzmau) [reply modules/installation-ibm-z-user-infra-machines-iso.adoc]: Is this a D.R.C issue? Andrea raised https://redhat-internal.slack.com/archives/C073SBBME1M/p1726663903236999 in the forum-docs-redhat Slack channel. The TOC on D.R.C is troublesome at times.

**Pull request merged**